### ISSUE TITLE
Add pillar specific renderer config option.

### ISSUE THREAD

**Issue by dnd** [saltstack/salt#13263]
Currently it is only possible to specify a renderer to render everything. In some instances, such as with the newly merged gpg renderer, the only option is the specify the renderers at the top of each file needing to use the renderer. If the master config has `renderer: jinja | yaml | gpg` the minion will always attempt to use the gpg renderer even though it has no ability to do so. This leads to an error as the dependencies are likely not present on the minion.

**Comment by vectorsigma** [comment]
FWIW, we really like the GPG rendering feature in Saltstack, so we're starting to roll it out farm wide to several thousand machines.  So the temporary fix of installing python-gnupg is only partially helpful, since RHEL 6 and RHEL 5 don't seem to ship that package.  We grabbed one from Fedora which seems to work, but that's an ugly hack.

TL;DR: Upvote from OpenX for fixing this bug. :+1:

**Comment by sastorsl** [comment]
@vectorsigma You can find python-gnupg for RHEL6 in https://fedoraproject.org/wiki/EPEL

**Comment by garrickp** [comment]
As an FYI, this is failing when using only the shebang as well:
```
$ head /srv/pillar/staging.sls
#!yaml|gpg
vagrant: false
dev: false
stage: true
[...]
$ sudo salt 'monitor-stage*' pillar.items
monitor-stage-1:
    ----------
    _errors:
        - Rendering SLS 'staging' failed. Please see master log for details.
[...]
$ tail /var/log/salt/master
[...]
2015-11-13 21:22:57,584 [salt.pillar      ][CRITICAL][826] Rendering SLS 'staging' failed, render error:
GPG unavailable
2015-11-13 21:22:57,588 [salt.pillar      ][CRITICAL][826] Pillar render error: Rendering SLS 'staging' failed. Please see master log for details.
[...]
$ grep gpg /etc/salt/master
$ salt --version
salt 2015.5.3 (Lithium)
$ sudo pip list | grep gnupg
python-gnupg (0.3.8)
$ gpg --version
gpg (GnuPG) 1.4.16
```

Upgraded and restarted everything.

**Comment by Sxderp** [comment]
Can we get this reopened? The issue has still not been resolved...
